app/controllers/suggestions.rb

```diff
@@ -0,0 +1,82 @@
+get "/suggestions" do
+  redirect "/"
+end
+
+get "/suggestions/new" do
+  current_user
+  redirect '/' unless authorized?(current_user)
+  erb :"suggestions/new"
+end
+
+post "/suggestions" do
+  current_user
+  redirect '/' unless authorized?(current_user)
+  @new_suggestion = Suggestion.new(user_id: current_user.id, title: params[:suggestion][:title], body: params[:suggestion][:body])
+  if @new_suggestion.save
+    redirect "/"
+  else
+    @errors = @new_suggestion.errors.full_messages
+    erb :"suggestions/new"
+  end
+end
+
+# Why does this route have to be below "/suggestions/new" to work
+get "/suggestions/:id" do
```

Comment:
Because `:id` catches whatever is after `/suggestions/`, including `new`. Sinatra is not smart enough to distinguish between integers and text.